app/main.py

```diff
@@ -1,8 +1,37 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people: list[dict]) -> list[Person]:
+    person_instances = [
+        Person(person_data["name"], person_data["age"])
+        for person_data in people
+    ]
+
+    for person_data in people:
+        if "wife" in person_data:
+            partner_name = person_data["wife"]
+            if (
+                partner_name is not None
+                and partner_name in Person.people
+            ):
```

Comment:
can you make only one condition? also checking `partner_name is not None` is unneccessary